### PR TITLE
ext_authz: Use 403 as default for denied response

### PR DESCRIPTION
**Pull request by dio**
Commit Message: Before this, when a gRPC server sends out `DeniedResponse` as a check response for a request but without setting the `HttpResponse.DeniedResponse.Status`, HTTP `ext_authz` filter translates that as `0` (empty/unknown HTTP status code). This patch makes sure we reply with a valid `403 Forbidden` HTTP status code (the current default status code for denied response).

Additional Description: When building a minimal gRPC auth server I found that it is a bit confusing for `CheckResponse_DeniedResponse` when you have already set `CheckResponse.Status` with `code.Code_PERMISSION_DENIED` but then you added `CheckResponse.HttpResponse`. In the current code, it is required to set `CheckResponse.HttpResponse.Status`, if not it will be sending Unknown (`0`) status code.

https://github.com/envoyproxy/envoy/blob/cc1d41e7ee9fbfb7ee3c8f73724cdc41d7c6bbb0/api/envoy/service/auth/v3/external_auth.proto#L49-L51

The above docs excerpt suggests that if we skip on specifying it, it will be 403.

Risk Level: Low
Testing: Added
Docs Changes: N/A
Release Notes: Added
Platform Specific Features: N/A

Signed-off-by: Dhi Aurrahman <dio@rockybars.com>

### CODE REVIEW

File: docs/root/version_history/current.rst [hunk 1]
```diff
@@ -69,7 +69,8 @@ Bug Fixes
 * cluster: finish cluster warming even if hosts are removed before health check initialization. This only affected clusters with :ref:`ignore_health_on_host_removal <envoy_v3_api_field_config.cluster.v3.Cluster.ignore_health_on_host_removal>`.
 * dynamic forward proxy: fixing a validation bug where san and sni checks were not applied setting :ref:`http_protocol_options <envoy_v3_api_msg_extensions.upstreams.http.v3.HttpProtocolOptions>` via :ref:`typed_extension_protocol_options <envoy_v3_api_field_config.cluster.v3.Cluster.typed_extension_protocol_options>`.
 * ext_authz: fix the ext_authz filter to correctly merge multiple same headers using the ',' as separator in the check request to the external authorization service.
-* ext_authz: the network ext_authz filter now correctly sets dynamic metdata returned by the authorization service for non-OK responses. This behavior now matches the http ext_authz filter.
+* ext_authz: fix the HTTP ext_authz filter to response with ``403 Forbidden`` when a gRPC auth server sends a denied check response with empty HTTP status code.
```
Comment (yanavlasov) [inline docs/root/version_history/current.rst:72]: s/response/respond

**Review by yanavlasov — COMMENTED** [review]
Should we make the status to not be required with this change?

/wait

**Comment by esmet** [comment]
Lets update the comment on status to make it extra clear that the default is 403. The language only sort of suggests this.

File: docs/root/version_history/current.rst (hunk 1 above)
Comment (dio) [reply docs/root/version_history/current.rst]: Thanks! Updated.

**Review by yanavlasov — DISMISSED** [review]

**Comment by dio** [comment]
Thanks, @yanavlasov, @esmet, and @esmet for the review!

**Comment by esmet** [comment]
Looks good, and I think it needs main merged in to fix the linux_x64 api_compat job.

**Comment by dio** [comment]
cc. @envoyproxy/senior-maintainers if you want to take a look. Thanks!

File: docs/root/version_history/current.rst [hunk 2]
```diff
@@ -78,7 +78,8 @@ Bug Fixes
 * cluster: finish cluster warming even if hosts are removed before health check initialization. This only affected clusters with :ref:`ignore_health_on_host_removal <envoy_v3_api_field_config.cluster.v3.Cluster.ignore_health_on_host_removal>`.
 * dynamic forward proxy: fixing a validation bug where san and sni checks were not applied setting :ref:`http_protocol_options <envoy_v3_api_msg_extensions.upstreams.http.v3.HttpProtocolOptions>` via :ref:`typed_extension_protocol_options <envoy_v3_api_field_config.cluster.v3.Cluster.typed_extension_protocol_options>`.
 * ext_authz: fix the ext_authz filter to correctly merge multiple same headers using the ',' as separator in the check request to the external authorization service.
-* ext_authz: the network ext_authz filter now correctly sets dynamic metdata returned by the authorization service for non-OK responses. This behavior now matches the http ext_authz filter.
+* ext_authz: fix the HTTP ext_authz filter to response with ``403 Forbidden`` when a gRPC auth server sends a denied check respond with empty HTTP status code.
```
Comment (mattklein123) [inline docs/root/version_history/current.rst:81]:
```suggestion
* ext_authz: fix the HTTP ext_authz filter to respond with ``403 Forbidden`` when a gRPC auth server sends a denied check response with an empty HTTP status code.
```

**Review by mattklein123 — CHANGES_REQUESTED** [review]
LGTM with small doc nit, thanks.

/wait

File: docs/root/version_history/current.rst (hunk 2 above)
Comment (dio) [reply docs/root/version_history/current.rst]: Thanks, @mattklein123. Updated.

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**